packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/ActivatePluginPostStartupActivity.kt

```diff
@@ -73,7 +74,7 @@ class ActivatePluginPostStartupActivity(
                     .addAction(OpenMongoDbPluginSettingsAction())
                     .notify(project)
 
-                settings.hasTelemetryOptOutputNotificationBeenShown = true
+                firstTimeTelemetry = true
```

Comment:
Something I want to bring attention to - When I first had a look at the original code of updating the PersistentStateComponent, I was a little amazed that simply setting a variable on a class instance will persist the state, was already kind of a magic(until I read how it works). But this feels a lot more magical that simply setting a local variable is supposed to update the persistent state. I do understand now how it works and the understandable side of me thinks of this being natural now but I think for someone new this might be confusing, something we should be aware of? Let me know what you think?

Reply:
Over the last few months, I developed a preference of explicit setter and getter functions over kotlin properties, helps me understand clearly what's gonna happen so for example. Imagine the above code being.
```
if (pluginSetting.firstTimeTelemetry) {
  // pluginSetting.firstTimeTelemetry is immutable
}

pluginSetting.updateAndPersist("firstTimeTelemetry", true)
```
(I understand and agree that this looks very 90'ish 😆)

Reply:
I personally prefer method calls, but I was testing property delegates because you can tell the compiler if something is mutable or not, whether using methods is not possible. You would need like two interfaces: one that only contains getters, and only that contains both getters and setters, and I feel this would be more cumbersome.

WDYT?

Reply:
It should be possible with methods as well to tell if something is mutable or not. A getter would only return an immutable value and one would use a setter for mutating that. I agree that it starts to become cumbersome when you intend to have more than a few properties and then corresponding getter/setter methods for those properties but then there we could use some generic method accepting key/value combo (not yet sure how that would work here in kotlin, so pardon me for bringing some TS concept here 😆).

I have no strong feelings against the delegate version, I was just a little surprised at first when I read in code that mutating a local variable in function also updates the state of the object from which it was retrieved from. But then I read the Java docs and it was clear how this works so I think at this point we can even go ahead and try this concept and see how it works for us?